doc/src/manual/conversion-and-promotion.md

```diff
@@ -134,52 +126,68 @@ actual Julia behaviour. This is the actual implementation in Julia:
 ```julia
 convert{T<:Real}(::Type{T}, z::Complex) = (imag(z)==0 ? convert(T,real(z)) :
                                            throw(InexactError()))
+```
+
+Which we can see in action:
+
+```jldoctest
+julia> convert(Bool, 0im)
+false
 
 julia> convert(Bool, 1im)
 ERROR: InexactError()
- in convert(::Type{Bool}, ::Complex{Int64}) at ./complex.jl:18
- ...
+Stacktrace:
+ [1] convert(::Type{Bool}, ::Complex{Int64}) at ./complex.jl:23
 ```
 
 ### [Case Study: Rational Conversions](@id man-rational-conversion)
 
 To continue our case study of Julia's `Rational` type, here are the conversions declared in [rational.jl](https://github.com/JuliaLang/julia/blob/master/base/rational.jl),
 right after the declaration of the type and its constructors:
 
-```julia
-convert{T<:Integer}(::Type{Rational{T}}, x::Rational) = Rational(convert(T,x.num),convert(T,x.den))
-convert{T<:Integer}(::Type{Rational{T}}, x::Integer) = Rational(convert(T,x), convert(T,1))
-
-function convert{T<:Integer}(::Type{Rational{T}}, x::AbstractFloat, tol::Real)
-    if isnan(x); return zero(T)//zero(T); end
-    if isinf(x); return sign(x)//zero(T); end
-    y = x
-    a = d = one(T)
-    b = c = zero(T)
-    while true
-        f = convert(T,round(y)); y -= f
-        a, b, c, d = f*a+c, f*b+d, a, b
-        if y == 0 || abs(a/b-x) <= tol
-            return a//b
-        end
-        y = 1/y
-    end
-end
-convert{T<:Integer}(rt::Type{Rational{T}}, x::AbstractFloat) = convert(rt,x,eps(x))
-
-convert{T<:AbstractFloat}(::Type{T}, x::Rational) = convert(T,x.num)/convert(T,x.den)
-convert{T<:Integer}(::Type{T}, x::Rational) = div(convert(T,x.num),convert(T,x.den))
+```jldoctest rational
+julia> convert{T<:Integer}(::Type{Rational{T}}, x::Rational) = Rational(convert(T,x.num),convert(T,x.den))
+convert (generic function with 1 method)
+
+julia> convert{T<:Integer}(::Type{Rational{T}}, x::Integer) = Rational(convert(T,x), convert(T,1))
+convert (generic function with 2 methods)
+
+julia> function convert{T<:Integer}(::Type{Rational{T}}, x::AbstractFloat, tol::Real)
```

Comment:
This method don't exist anymore. Should probably mimic [this](https://github.com/JuliaLang/julia/blob/master/base/rational.jl#L80) instead?